tools/oshinko-deploy.sh

```diff
@@ -55,71 +81,66 @@ while getopts :s:w:p:u:h opt; do
     esac
 done
 
-if [ -z "$PROJECT" ]
-then
-    echo "project name not set, using default value"
-    PROJECT=$DEFAULT_OPENSHIFT_PROJECT
-fi
-
 if [ -z "$OS_USER" ]
 then
     echo "user not set, using default value"
     OS_USER=$DEFAULT_OPENSHIFT_USER
 fi
 
+if [ -z "$PROJECT" ]
+then
+    echo "project name not set, using default value"
+    PROJECT=$DEFAULT_OPENSHIFT_PROJECT
+fi
+
 if [ -z "$SPARK_IMAGE" ]
 then
     SPARK_IMAGE=$DEFAULT_SPARK_IMAGE
 fi
 
-oc login -u system:admin
-oc project default
-REGISTRY=$(oc get service docker-registry --no-headers=true | awk -F ' ' '{print $2":"$4}' | sed "s,/TCP$,,")
-
-# reset back to the default development account
-oc login -u $OS_USER
-oc project $PROJECT
+if [ -z "$WEB_IMAGE" ]
+then
+    WEB_IMAGE=$DEFAULT_OSHINKO_WEB_IMAGE
+fi
 
-# Wait for the registry to be fully up
-r=1
-while [ $r -ne 0 ]; do
-    docker login -u $(oc whoami) -e "jack@jack.com" -p $(oc whoami -t) $REGISTRY
-    r=$?
-    sleep 1
-done
+if [ -z "$REST_IMAGE" ]
+then
+    REST_IMAGE=$DEFAULT_OSHINKO_REST_IMAGE
+fi
 
-docker tag oshinko-rest $REGISTRY/$PROJECT/oshinko-rest
-docker push $REGISTRY/$PROJECT/oshinko-rest
-docker tag oshinko-webui $REGISTRY/$PROJECT/oshinko-webui
-docker push $REGISTRY/$PROJECT/oshinko-webui
-docker tag radanalytics-pyspark $REGISTRY/$PROJECT/radanalytics-pyspark
-docker push $REGISTRY/$PROJECT/radanalytics-pyspark
-
-# check to see if we have a local copy of the spark image
-# otherwise, pull it before tagging for oshinko
-local_spark_image=$(docker images -q $SPARK_IMAGE)
-if [ -z "$local_spark_image" ]
+if [ -n "$OS_ALLINONE" ]
 then
-    docker pull $SPARK_IMAGE
+    if [ -n "$OS_CLUSTER" ]
+    then
+        echo "You have requested an all-in-one deployment AND specified a cluster address."
+        echo "Please choose one of these options and restart."
+        exit 1
+    else
+        oc cluster up
+    fi
 fi
-docker tag $SPARK_IMAGE $REGISTRY/$PROJECT/oshinko-spark
-docker push $REGISTRY/$PROJECT/oshinko-spark
 
-# set up the oshinko service account
+oc login $OS_CLUSTER -u $OS_USER
+
+oc new-project $PROJECT
+
```

Comment:
ah, it must be the new-app that's failing then

Error from server: project "myproject" already exists
serviceaccount "oshinko" created
template "oshinko" created
error: no match for "oshinko"

The 'oc new-app' command will match arguments to the following types: